queries/slint/highlights.scm

```diff
@@ -1,121 +1,127 @@
-(identifier) @variable
-
-(type_identifier) @type
-
 (comment) @comment @spell
 
-(int_literal) @number
+; Different types:
+(string_value) @string @spell
+
+(escape_sequence) @string.escape
 
-(float_literal) @number.float
+(color_value) @constant
 
-(string_literal) @string
+[
+  (children_identifier)
+  (easing_kind_identifier)
+] @constant.builtin
 
-(function_identifier) @function
+(bool_value) @boolean
 
 [
-  (image_macro)
-  (children_macro)
-  (radial_grad_macro)
-  (linear_grad_macro)
-] @function.macro
+  (int_value)
+  (length_value)
+  (physical_length_value)
+  (duration_value)
+  (angle_value)
+  (relative_font_size_value)
+] @number
 
-(call_expression
-  function: (identifier) @function.call)
+(simple_identifier) @property
 
-(call_expression
-  function:
-    (field_expression
-      field: (identifier) @function.call))
+(purity) @type.qualifier
 
-(vis) @keyword.import
+(function_visibility) @type.qualifier
 
-(units) @type
+(property_visibility) @type.qualifier
 
-(array_literal
-  (identifier) @type)
+(animate_option_identifier) @keyword
 
-(transition_statement
-  state: (identifier) @variable.member)
+[
+  (float_value)
+  (percent_value)
+] @number.float
 
-(state_expression
-  state: (identifier) @variable.member)
+(builtin_type_identifier) @type.builtin
 
-(struct_block_definition
-  (identifier) @variable.member)
+(reference_identifier) @variable.builtin
 
-; (state_identifier) @variable.member
-[
-  "in"
-  "for"
-] @keyword.repeat
+(type
+  [
+    (type_list)
+    (user_type_identifier)
+    (anon_struct_block)
+  ]) @type
 
-"@" @keyword
+(user_type_identifier) @type
 
 [
-  "import"
-  "from"
-] @keyword.import
+  (comparison_operator)
+  (mult_prec_operator)
+  (add_prec_operator)
+  (unary_prec_operator)
+  (assignment_prec_operator)
+] @operator
 
-[
-  "if"
-  "else"
-] @keyword.conditional
+; Functions and callbacks
+(argument) @variable.parameter
 
-[
-  "root"
-  "parent"
-  "duration"
-  "easing"
-] @variable.builtin
+(function_call) @function.call
 
-[
-  "true"
-  "false"
-] @boolean
+; definitions
+(callback
+  name: (_) @function)
 
-[
-  "struct"
-  "property"
-  "callback"
-  "in"
-  "animate"
-  "states"
-  "when"
-  "out"
-  "transitions"
-  "global"
-] @keyword
+(component
+  id: (_) @variable)
 
-[
-  "black"
-  "transparent"
-  "blue"
-  "ease"
-  "ease_in"
-  "ease-in"
-  "ease_in_out"
-  "ease-in-out"
-  "ease_out"
-  "ease-out"
-  "end"
-  "green"
-  "red"
-  "red"
-  "start"
-  "yellow"
-  "white"
-  "gray"
-] @constant.builtin
+(enum_definition
+  name: (_) @type)
+
+(function_definition
+  name: (_) @function)
+
+(property
+  name: (_) @property)
 
-; Punctuation
+(struct_definition
+  name: (_) @type)
+
+(typed_identifier
+  name: (_) @variable)
+
+(typed_identifier
+  type: (_) @type)
+
+(binary_expression
+  op: (_) @operator)
+
+":=" @operator
+
+(unary_expression
+  op: (_) @operator)
+
+(if_statement
+  "if" @keyword.conditional)
+
+(if_statement
+  ":" @punctuation.delimiter)
```

Comment:
I would say don't make ":" specific, it's missing in a few cases, and the ternary capture below it will override it anyways. You're also missing => (and maybe some more?)
```suggestion
[
  ":"
  "=>"
] @punctuation.delimiter
```

Reply:
Is `=>` actually an operator? I don't really know slint

Reply:
You're also missing ":" here, so it's easier to just not have it be specifically under another node, but rather captured directly.

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/29718261/8e3d20a1-c7ba-4b62-b06b-f3761529f189) here it's not captured

Reply:
As well as here you didn't capture =>

![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/29718261/33baea65-2590-44e1-9cdc-7531a6c14b5e)